fem/complex_fem.cpp

```diff
@@ -1204,17 +1204,28 @@ ParSesquilinearForm::FormLinearSystem(const Array<int> &ess_tdof_list,
       });
       // Modify offdiagonal blocks (imaginary parts of the matrix) to conform
       // with standard essential BC treatment
-      ess_tdof_list.HostRead();
       if (A_i.Type() == Operator::Hypre_ParCSR)
       {
          HypreParMatrix * Ah;
          A_i.Get(Ah);
          hypre_ParCSRMatrix *Aih = *Ah;
+#ifndef HYPRE_USING_CUDA
+         ess_tdof_list.HostRead();
          for (int k = 0; k < n; k++)
          {
             const int j = ess_tdof_list[k];
             Aih->diag->data[Aih->diag->i[j]] = 0.0;
          }
+#else
+         // FIXME: need Ah->HypreReadWrite() since Ah will be modified.
```

Comment:
Is this still needed?

Reply:
It is not causing any problems at this point, as far as I know, however, I still think this is needed for correctness. We can probably leave it "as is" for now but it should be fixed at some point.